interop-twitter/jvm/src/main/scala/scalaz/zio/interop/twitter.scala

```diff
@@ -33,4 +34,25 @@ package object twitter {
         }
       }
   }
+
+  implicit class RuntimeOps[R](private val runtime: Runtime[R]) extends AnyVal {
+    def unsafeRunToTwitterFuture[E <: Throwable, A](io: ZIO[R, E, A]): Future[A] =
+      runtime.unsafeRun {
+        io.fork.flatMap { f =>
+          val p                              = Promise[A]()
+          def failure(cause: Cause[E]): Unit = p.setException(cause.squash)
+          def success(value: A): Unit        = p.setValue(value)
+
+          p.setInterruptHandler {
+            case _ =>
+              val _ = runtime.unsafeRunSync(f.interrupt)
+          }
+
+          f.await
+            .flatMap(exit => Task.effect(exit.fold(failure, success)))
+            .fork
```

Comment:
Instead of forking (again) here, you could use `unsafeRunAsync` at the top level to avoid the fork.

Reply:
updated